packages/app-project/src/screens/ClassifyPage/components/RecentSubjects/components/SubjectPreview/SubjectPreview.js

```diff
@@ -0,0 +1,83 @@
+import counterpart from 'counterpart'
+import { array, bool, number, shape, string } from 'prop-types'
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import { Anchor, Box } from 'grommet'
+import { Media } from '@zooniverse/react-components'
+import { CollectionsButton, FavouritesButton, TalkLink } from './components'
+import CollectionsModal from '../../../../components/CollectionsModal'
+
+import en from './locales/en'
+
+counterpart.registerTranslations('en', en)
+
+function SubjectPreview ({ height, recent, slug, width }) {
+  const subjectURLs = recent.locations.map(location => Object.values(location)[0])
+  const subjectURL = subjectURLs[0]
+  const [ checked, setChecked ] = useState(recent.favorite)
```

Comment:
Do we have any examples of tests with hooks so far? I was trying to write one for `useRouter` the other day and couldn't get it working.

Reply:
Not yet.

Reply:
Here, I think a test would click `FavouritesButton` and expected its `checked` attribute to change from `false` to `true`. See the counter example here:
https://reactjs.org/docs/hooks-faq.html#how-to-test-components-that-use-hooks

>From React’s point of view, a component using Hooks is just a regular component. If your testing solution doesn’t rely on React internals, testing components with Hooks shouldn’t be different from how you normally test components.

Reply:
I've added those tests, so we now have examples of tests that use hooks.

Sounds like enzyme only partially supports hooks right now: https://blog.carbonfive.com/2019/08/05/shallow-testing-hooks-with-enzyme/

From reading up a bit around that, it may be a React issue with shallow rendering not updating the component via `useEffect`. The workaround might be to use `mount` to test anything other than `useState`.